src/Nethermind/Nethermind.Consensus.Clique/CliqueSealer.cs

```diff
@@ -64,7 +66,20 @@ public CliqueSealer(ISigner signer, ICliqueConfig config, ISnapshotManager snaps
 
             // Sign all the things!
             Hash256 headerHash = SnapshotManager.CalculateCliqueHeaderHash(header);
-            Signature signature = _signer.Sign(headerHash);
+            Signature signature;
+            if (_signer.CanSignHeader)
+            {
+                BlockHeader clone = header.Clone();
+                int extraSeal = 65;
+                clone.ExtraData = clone.ExtraData.Slice(0, clone.ExtraData.Length - extraSeal);
+                clone.Hash = headerHash;
```

Comment:
This looks like copy of what `SnapshotManager.CalculateCliqueHeaderHash` is doing, maybe this could be part of `SnapshotManager`?

Reply:
created a small helper function.